### ISSUE TITLE
RpcCatalog.Register not giving an exception when a procedure already exists

### ISSUE THREAD

**Issue by BramVader** [Code-Sharp/WampSharp#255]
When I have two clients register a procedure having the same name and both have WampInvokePolicy.Single, I would expect that the registration of the second client fails with an exception ("wamp.error.procedure_already_exists"). That is not the case. The returned Task<IAsyncDisposable> just completes like nothing happened.
When calling the procedure the one that registered first gets the call.

**Comment by darkl** [comment]
Can you please share a simple code that reproduces this?
[This](https://gist.github.com/darkl/66aa282e6444026402ba32ab8b2c19e9) code example works fine for me (running the client exe twice and getting the exception for the second client).

Elad

**Comment by BramVader** [comment]
Hi Elad,

Thanks for your response.

I tested your Client.cs-code and when I run it twice I get the exception. When I run one instance of your client and then my own code (or vice versa), the exception doesn't occur. In both cases your client is able to 'hijack' the procedure (so it runs the ArgLenOperation instead of my own procedure).

But I found out what's going on. It's because I'm using WampMatchPattern.Wildcard for all my procedures, because _some_ of them have wildcards (like: myprefix.v2.person..setage), others don't (like: myprefix.v2.myservice1.addperson).

When I set it back to WampMatchPattern.Default / .Exact it works like expected and I get the exception on either side (depending on which one starts second). However, this behavior is not what I would expect. Even when using wildcards, this procedure (having WampInvokePolicy.Single) I register must be the only one that should exist. 

I know there can be corner cases though _when routing actual calls_, like client A has registered myprefix.v2.person..setage and client B has registered myprefix.v2.person.Piet.setage, but as long as they have a different uri, they should be treated as different registrations.

But I guess this is a router problem and not a WampSharp problem (I'm using the crossbar.io's Autobahn router Docker-container by the way).
